arithmetization/src/test/java/net/consensys/linea/zktracer/module/blsdata/BlsPairingCheckTest.java

```diff
@@ -0,0 +1,125 @@
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import net.consensys.linea.UnitTestWatcher;
+import net.consensys.linea.reporting.TracerTestBase;
+import net.consensys.linea.testing.BytecodeCompiler;
+import net.consensys.linea.testing.BytecodeRunner;
+import net.consensys.linea.testing.ToyAccount;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(UnitTestWatcher.class)
+public class BlsPairingCheckTest extends TracerTestBase {
+
+  @ParameterizedTest
+  @MethodSource("blsPairingCheckSource")
+  void testBlsPairingCheckTest(
+      List<String> smallPoints, List<String> largePoints, TestInfo testInfo) {
+    checkArgument(smallPoints.size() == largePoints.size());
+    Bytes input =
+        IntStream.range(0, smallPoints.size())
+            .mapToObj(
+                i ->
+                    Bytes.concatenate(
+                        Bytes.fromHexString(smallPoints.get(i)),
+                        Bytes.fromHexString(largePoints.get(i))))
+            .reduce(Bytes.EMPTY, Bytes::concatenate);
+
+    BytecodeCompiler program = BytecodeCompiler.newProgram(chainConfig);
+
+    final Address codeOwnerAddress = Address.fromHexString("0xC0DE");
+    final ToyAccount codeOwnerAccount =
+        ToyAccount.builder()
+            .balance(Wei.of(0))
+            .nonce(1)
+            .address(codeOwnerAddress)
+            .code(input)
+            .build();
+
+    // First place the parameters in memory
+    // Copy to targetOffset the code of codeOwnerAccount
+    program
+        .push(codeOwnerAddress)
+        .op(OpCode.EXTCODESIZE) // size
+        .push(0) // offset
+        .push(0) // targetOffset
+        .push(codeOwnerAddress) // address
+        .op(OpCode.EXTCODECOPY);
+
+    // Do the call
+    program
+        .push(32) // retSize
+        .push(input.size()) // retOffset
+        .push(input.size()) // argSize
+        .push(0) // argOffset
+        .push(Address.BLS12_PAIRING) // address
+        .push(Bytes.fromHexStringLenient("0xFFFFFFFF")) // gas
+        .op(OpCode.STATICCALL);
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(program.compile());
+    bytecodeRunner.run(List.of(codeOwnerAccount), chainConfig, testInfo);
+
+    if (isPostPrague(fork)) {
+      final boolean failureIsExpected =
+          smallPoints.stream().anyMatch(p -> !p.equals(VALID_G1_POINT))
+              || largePoints.stream().anyMatch(p -> !p.equals(VALID_G2_POINT));
+      final BlsData blsdata = (BlsData) bytecodeRunner.getHub().blsData();
+      assertEquals(failureIsExpected, blsdata.blsDataOperation().mext());
```

Comment:
We have an issue about using proper names instead of the spec shorthands (`mint` / `mext` / etc ...), see https://github.com/Consensys/linea-tracer/issues/2341